File: 프로그래머스/1/135808. 과일 장수/과일 장수.java
import java.util.*;
class Solution {
    public int solution(int k, int m, int[] score) {
        int ret = 0;
        Integer[] temp = new Integer[score.length];
        for(int i = 0; i < score.length; i++){
            temp[i] = score[i];
        }
        Arrays.sort(temp, Collections.reverseOrder());
        int j = 0;
        for(int i = 0; i < temp.length; i++){
            j++;
            if(j == m) {
                ret += temp[i] * j;
                j = 0;
            }
        }
        
        return ret;
    }
}

File: 프로그래머스/1/135808. 과일 장수/과일 장수.py
def solution(k, m, score):
    temp = []
    score.sort(reverse=True)
    ret = 0
    for s in score:
        temp.append(s)
        if len(temp) == m:
            ret += temp[-1] *m
            temp = []
    return ret
    
"""
k점 최상, 한상자에m개씩
상자단위로 판매 남는거 버림
내림차순으로 정렬하고 비싼거 끼리 몰아서 넣고 그다음 보내기

"""